forms-flow-data-analysis-api/README.md

```diff
@@ -6,43 +6,32 @@
+4. Finally, we choose DistilBERT for sentiment analysis. Compared to all other models DistilBERT is faster, smaller, lighter and gives better accuracy.
 
-4. Once the data was prepared, a variety of techniques for identifying topics and sentiment analysis was tried
-out like Latent Semantic Analysis, LSTMs, BERTs. Finally, the ML model was built using a two-stage pipeline process
-to identify entities and their associated sentiment. We use a named entity recognition model(NER) to
-identify the topics, and further sentiment analysis is being done for individual entities.
+4. Model we will use is from the Transformer library, we need to install it using python package manager(pip). Now we need to apply Auto tokenizer to use pre-trained tokenizers. For more details refer [HuggingFace Tokenizers](https://huggingface.co/docs/transformers/main_classes/tokenizer).
+Next  prepare the data according to the format needed for the BERT model:
+* Input IDs – The input ids are often the only required parameters to be passed to the model as input. 
+* Attention mask – Attention Mask is used to avoid performing attention on padding token indices. Mask value can be either 0 or 1, 1 for tokens that are NOT MASKED, 0 for MASKED tokens.
 
+5. Transformers provides a Trainer class optimized for training. Transformers models, making it easier to start training without manually writing code. Start by loading your model and specify the number of expected labels.
 
-5. The output of our model for the model using our ML model for a sample input is as shown below:
+6. Next, create a Training Arguments class which contains all the hyperparameters you can tune as well as flags for activating different training options. You can start with the default training hyperparameters. Here we use:
+* learning_rate=5.691013656357132e-06,
+* num_train_epochs=5,
+* per_device_train_batch_size=8,
+* seed=36,
+* optim="adamw_torch"
 
-> text="awesome location and great staff. Staff provided excellent service."
+7. Trainer does not automatically evaluate model performance during training. You will need to pass Trainer a function to compute and report metrics. The Datasets library provides a simple accuracy function you can load with the load_metric function.
 
-```
-{
-  'sentiment': {'location': 'positive', 'facility': 'positive'},
-  'overall_sentiment': 'positive'
-}
-```
+8. Create a Trainer object with your model, training arguments, training and test datasets, and evaluation function. Then fine-tune your model by calling train(). This model reaches an accuracy of 86.
 
```

Comment:
MODEL_ID is the env for selecting the model for the sentiment analysis component, user can fine-tune the model on their own dataset and use the model by providing the hugging face model id